mypy error for HKT example from docs

# Bug report

I was going through your documentation and stumbled upon an issue in HKT example: https://returns.readthedocs.io/en/latest/pages/hkt.html#id1.

## What's wrong

Trying out your `to_str` definition:

```python
from typing import TypeVar
from returns.maybe import Maybe
from returns.primitives.hkt import Kind1, kinded
from returns.interfaces.container import Container1, ContainerN

T = TypeVar("T", bound=Container1)

@kinded
def to_str(container: Kind1[T, int]) -> Kind1[T, str]:
    return container.map(str)

reveal_type(to_str(Maybe.from_value(1)))
```

Produces `mypy` error: `Value of type variable "T" of "to_str" cannot be "Maybe[Any]"  [type-var]`.

Playing around, I found, that replacing `Container1` with `ContainerN` fixes the issue (why?).

```python
T = TypeVar("T", bound=ContainerN)
```

## How is that should be

Examples from the documentation should work :)

I'm not sure whether this is a defect, maybe I'm doing something wrong, but before submitting this one I've rechecked that my code matches the docs closely.

## System information

- `python` version: 3.11.1
- `returns` version: 0.20.0
- `mypy` version: 1.3.0
  - I'm using config from here: https://github.com/wemake-services/wemake-python-styleguide/blob/master/styles/mypy.toml

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -33,8 +33,7 @@ jobs:
 
     - name: Install poetry
       run: |
-        curl -sSL \
-          "https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py" | python
+        curl -sSL "https://install.python-poetry.org" | python
 
         # Adding `poetry` to `$PATH`:
         echo "$HOME/.poetry/bin" >> $GITHUB_PATH

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,13 @@ incremental in minor, bugfixes only are patches.
 See [0Ver](https://0ver.org/).
 
 
+## 0.20.1 WIP
+
+### Bugfixes
+
+- Fixed HKT `to_str` example in documentation
+
+
 ## 0.20.0
 
 ### Features

--- a/docs/pages/hkt.rst
+++ b/docs/pages/hkt.rst
@@ -184,13 +184,13 @@ Let's see how it works:
 .. code:: python
 
   >>> from returns.primitives.hkt import Kind1
-  >>> from returns.interfaces.container import Container1
+  >>> from returns.interfaces.container import ContainerN
   >>> from typing import TypeVar
 
-  >>> T = TypeVar('T', bound=Container1)
+  >>> T = TypeVar('T', bound=ContainerN)
 
-  >>> def to_str(arg: Kind1[T, int]) -> Kind1[T, str]:
-  ...   ...
+  >>> def to_str(container: Kind1[T, int]) -> Kind1[T, str]:
+  ...     return container.map(str)
 
 Now, this will work almost correctly!
 Why almost? Because the revealed type will be ``Kind1``.
@@ -210,8 +210,8 @@ The final solution is to decorate ``to_str`` with ``@kinded``:
   >>> from returns.primitives.hkt import kinded
 
   >>> @kinded
-  ... def to_str(arg: Kind1[T, int]) -> Kind1[T, str]:
-  ...   ...
+  ... def to_str(container: Kind1[T, int]) -> Kind1[T, str]:
+  ...     return container.map(str)
 
 Now, it will be fully working:
 
@@ -222,7 +222,7 @@ Now, it will be fully working:
 
 And the thing about this approach is that it will be:
 
-1. Fully type-safe. It works with correct interface ``Container1``,
+1. Fully type-safe. It works with correct interface ``ContainerN``,
    returns the correct type, has correct type transformation
 2. Is opened for further extension and even custom types
 

